src/portal/src/app/project/project-detail/project-detail.component.ts

```diff
@@ -188,10 +219,40 @@ export class ProjectDetailComponent implements OnInit {
   isDefaultTab(tab, index) {
     return this.route.snapshot.children[0].routeConfig.path !== tab.linkName && index === 0;
   }
+
   isTabLinkInOverFlow() {
     return this.tabLinkNavList.some(tab => {
       return tab.tabLinkInOverflow && this.route.snapshot.children[0].routeConfig.path === tab.linkName;
     });
   }
 
+  @HostListener('window:resize', ['$event'])
+  onResize(event) {
+    if (this.previousWindowWidth) {
+       // down size
+      if (this.previousWindowWidth > event.target.innerWidth) {
+          this._subject.next(DOWN);
+      } else { // up size
+          this._subject.next(UP);
+      }
+    }
+    this.previousWindowWidth = event.target.innerWidth;
+  }
+
+  resetTabsForDownSize(): void {
+    this.tabLinkNavList.filter(item => !item.tabLinkInOverflow).forEach((item, index) => {
+      const right: number = window.innerWidth - document.getElementById('project-' + item.linkName).getBoundingClientRect().right;
+      if (right < SHOW_ELLIPSIS_WIDTH) {
+        this.tabLinkNavList[index].tabLinkInOverflow = true;
+      }
+    });
+  }
+  resetTabsForUpSize() {
+    for ( let i = 0; i < this.tabLinkNavList.length; i++) {
+      this.tabLinkNavList[i].tabLinkInOverflow = false;
+    }
+    this.cdf.detectChanges();
```

Comment:
Need add comment here to indicate why need this detectChange.

Reply:
Fixed

Reply:
> Need add comment here to indicate why need this detectChange.

Fixed